### PR TITLE
Fix Storybook Theme page

### PR DESCRIPTION
**Pull request by robinmetral**
## Purpose

As title suggests.

## Approach and changes

- Reintroduces `Type` component, mistakenly removed in #1392 
- Addresses leftover deprecation warnings specific to the Theme page (because the page is using several custom Storybook components)

_Note: I'm also noticing that the theme information on the page is not fully up-to-date (for example z-indices). We'll look into tying this closer to the actual theme to avoid mismatches in the future_

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: docs/features/theme.stories.mdx [hunk 1]
```diff
@@ -16,12 +15,8 @@ import { Headline, SubHeadline, Body } from '../../packages/circuit-ui';
 
 # Theme
 
-<Intro>
-
 The theme used throughout Circuit UI comes from SumUp's design token library, [`@sumup/design-tokens`](Packages/design-tokens), which is a required peer dependency of `@sumup/circuit-ui`. Refer to its [documentation](Packages/design-tokens) for usage instructions.
 
-</Intro>
-
```
Comment (connor-baer) [inline docs/features/theme.stories.mdx:24]: Why did you remove the `Intro` here? I preferred the previous styling 😕

Comment (robinmetral) [reply docs/features/theme.stories.mdx:24]: Because nested `<p>` tags is invalid markup (and by extension a WCAG violation). I'm happy to look into a better styling solution for this `Intro` component like I mentioned [here](https://github.com/sumup-oss/circuit-ui/pull/1392#discussion_r794882086), but I think that we should start with strong foundations and valid HTML 🙂 

Screenshot 2022-01-29 at 16 07 32

Comment (robinmetral) [reply docs/features/theme.stories.mdx:24]: Actually it's pretty straightforward—the `Intro` component should use the new `BodyLarge` 🙂

Comment (robinmetral) [reply docs/features/theme.stories.mdx:24]: Hm but actually, the problem is that in MDX if you do this:

```mdx
<Intro>

Intro text here

</Intro>
```

The text will be wrapped in a paragraph:

```html
<p> <!-- 👈 this is the `Intro` component -->
  <p>Intro text here</p>
</p>
```

---

And in contrast if you omit the line breaks:

```mdx
<Intro>Intro text here</Intro>
```

...then you can't include markdown links in the intro text (`[link](url)`), because it will be considered HTML. So in our case we could still link to the ^Packages/Design Tokens` page but we'd have to make this a "real" link which would cause a full page load

Comment (connor-baer) [reply docs/features/theme.stories.mdx:24]: How about rendering the Intro component as a `div`?

```html
<div> <!-- 👈 this is the `Intro` component -->
  <p>Intro text here</p>
</div>
```

Comment (robinmetral) [reply docs/features/theme.stories.mdx:24]: Yes this would work but I think that the child p styles would override the parent div styles (so the text would be typography.body.one). I can find a solution on Monday though (there's always !important 🙈)
